### DISCUSSION TITLE
Different approach

### DISCUSSION THREAD

**Discussion by marioverhaeg** [thomase1234/esphome-fake-xemex-csmb#16] [General]
Hi Thomas,

As I don't want to use ESP modules I went for a different solution. I've bought a Modbus (RS485) to TCP converter. I've build a .NET service that listens to the modbus messages and responds on the requests from the wallbox. 

So far so good: I'm seeing modbus messages from the wallbox and the CSMB module. I've also interpreted them and can generate responses. I've checked the checksum and double check if my responses are in the correct (HEX) format by generating exactly the same values as generated by the CSMB. 

However, the charging speed of the wallbox doesn't match my expectations. I'm noticing that my messages do have an impact on the charging speed, but not the expected impact. For the last 20 minutes I've been sending 5A messages on all three phases (I have a 3x25A input) but the car only started charging after 18 minutes, even though the values have not changed. I would also expect a full power (16x3A charge) but instead the charging speed goes up and down, ranging from 9 to 5 to 0kW. 

Does this sound familiar to you? Do you have any tips you can give?

Thanks!
Mario

**Comment by thomase1234** [comment]
Hi Mario,

I can't explain the behavior that you experience with your 3Phase connection. 
From my understanding, charging should start immediately when you imitate a 5A load on each of the phases. 5A means that there is plenty remaining to get the charging started.

Here is a post from a user that also tries to influence the charging speed with a 3 phase grid connection. That user is also struggling to get it working.
[Home assistant forum](https://community.home-assistant.io/t/my-new-motion-integration-ev-charging-from-shell-newmotion/369593/181?u=thomase1234)

Best regards and good luck!

**Reply by marioverhaeg** [reply]
Thank you Thomas. Based on another discussion on a Dutch forum I've reached out to Shell and they will check if the DPM is actually working as designed. There might be another issue with my installation. When the DPM works as designed, I can continue to try to influence it.
